mtags/src/main/scala/scala/meta/internal/pc/ScaladocCompletion.scala

```diff
@@ -0,0 +1,199 @@
+      // """
+      val builder = new StringBuilder()
+
+      val hasConstructor = associatedDef.symbol.primaryConstructor.isDefined && !associatedDef.symbol.isAbstractClass
+      val hasReturnValue =
+        associatedDef.symbol.isMethod &&
+          !(associatedDef.symbol.asMethod.returnType.finalResultType
+            =:= typeOf[Unit])
+
+      // newline after `/**`
+      builder.append("\n")
+
+      // * $0 <- move cursor here.
+      builder.append(s"${indent}*")
+      if (clientSupportsSnippets) builder.append(" $0\n")
+      else builder.append("\n")
+
+      // add empty line if there's parameter or "@return" or "@constructor" line.
+      if (params.nonEmpty || hasReturnValue || hasConstructor)
+        builder.append(s"${indent}*\n")
+
+      // * @constructor
+      if (hasConstructor) builder.append(s"${indent}* @constructor\n")
+
+      // * @param p1
+      // * @param p2
+      params.foreach(param => builder.append(s"${indent}* @param ${param}\n"))
+
+      // * @return
+      // */
+      if (hasReturnValue) builder.append(s"${indent}* @return\n")
+      builder.append(s"${indent}*/")
+
+      List(
+        new TextEditMember(
+          "Scaladoc Comment",
+          new l.TextEdit(
+            editRange,
+            builder.toString()
+          ),
+          completionsSymbol(associatedDef.name.toString()),
+          label = Some("/** */"),
+          detail = Some("Scaladoc Comment")
+        )
+      )
+    }
+
+    // Infers the indentation at the completion position by counting the number of leading
+    // spaces in the line.
+    // For example:
+    // |"""
+    // |object A {
+    // |  /**<COMPLETE> // inferred indent is 4 spaces
+    // |  def foo(x: Int) = ???
+    // |}
+    // |"""
+    private def inferIndent(pos: Position): String = {
+      if (metalsConfig.snippetAutoIndent()) {
+        ""
+      } else {
+        // 3 is the length of "/**$" <- $ is a cursor
+        " " * (pos.column - 4)
+      }
+    }
+
+    /**
+     * Returns the parameter names of the given memberDef.
+     *
+     * @param memberDef The memberDef to construct scaladoc.
+     */
+    private def getParams(memberDef: MemberDef): List[String] = {
+      memberDef match {
+        case defdef: DefDef =>
+          defdef.symbol.paramss.flatten
+            .filter(!_.isSynthetic)
+            .map(_.name.toString())
+        case clazz: ClassDef =>
+          // If the associated def is a class definition,
+          // retrieve the constructor from the class, and caluculate the lines
+          // from the constructor definition instead.
+          clazz.symbol.primaryConstructor.paramss.flatten
+            .filter(sym =>
+              // FIXME: All synthetic params should be filtered out by checking SYNTHETIC flag.
+              // evidence params for classs constructor look like they don't have SYNTHETIC flag.
+              !sym.isSynthetic &&
+                !sym.name.startsWith(nme.EVIDENCE_PARAM_PREFIX)
```

Comment:
I believed that we can filter out all synthetic parameter like `evidence$1: Ordering[T]` which is appended here https://github.com/scala/scala/blob/ba9701059216c629410f4f23a2175d20ad62484b/src/compiler/scala/tools/nsc/ast/parser/TreeBuilder.scala#L135
But only with `isSynthetic` filter, I couldn't filter out `evidence$1` from params and [this test](https://github.com/scalameta/metals/pull/1250/files#diff-a92ad42d0d97b9389b28fec32531915dR267) failed 🤔 

I think I have to dig deeper about this phenomenon, but I just patched it with filtering by `EVIDENCE_PARAM_PREFIX` this for now.

Reply:
I'm not still 100% sure, but maybe this happens because the scalac type checker doesn't keep tree modifiers in sync with symbol flags for a class constructor. (actually, there's only `IMPLICIT` flag remains on the symbol of `evidence` on class constructor).

Maybe I can create an issue on `scala/bug` and fix (or maybe feature?) this,  but let me keep this workaround and note this behavior on a comment for now :)

---


- when a symbol for a method signature is created, modifiers that are defined as [ValueParameterFlags](https://github.com/scala/scala/blob/8514c3066ac8f71e676369e6163a1b2e6303f965/src/reflect/scala/reflect/internal/Flags.scala#L303-L307) survive during type checking.
  - Assigning modifiers to value parameters [here]( https://github.com/scala/scala/blob/4f04e38f1d0a74048b4ea350e7c7ff377b6fa3d4/src/compiler/scala/tools/nsc/typechecker/Namers.scala#L1332-L1336)
- but, it looks like type-checker doesn't use `ValueParameterFlags` as modifiers mask for a class constructor. Just assigning [inConstructorFlags]( https://github.com/scala/scala/blob/4f04e38f1d0a74048b4ea350e7c7ff377b6fa3d4/src/compiler/scala/tools/nsc/typechecker/Namers.scala#L409) (I'm not 100% sure...), and all modifiers from the tree don't remain in symbol definition.

FYI: The PR that makes `evidence` (for method definition symbol) SYNTHETIC is here https://github.com/scala/scala/pull/2132